dali/test/python/test_operator_numpy_reader.py

```diff
@@ -22,28 +22,72 @@
 import tempfile
 
 class NumpyReaderPipeline(Pipeline):
-    def __init__(self, path, batch_size, path_filter="*.npy", num_threads=1, device_id=0, num_gpus=1):
+    def __init__(self, path, batch_size, path_filter="*.npy",
+                 num_threads=1, device_id=0, num_gpus=1,
+                 slab_anchor=None, slab_shape=None, io_size=0):
         super(NumpyReaderPipeline, self).__init__(batch_size,
                                                   num_threads,
                                                   device_id)
+
+        self.input = None
+        if (slab_anchor is not None) and (slab_shape is not None):
+            self.input = ops.NumpyReader(file_root = path,
+                                         file_filter = path_filter,
+                                         anchor = slab_anchor,
+                                         shape = slab_shape,
+                                         shard_id = device_id,
+                                         num_shards = num_gpus,
+                                         target_io_bytes = io_size)
+        else:
+            self.input = ops.NumpyReader(file_root = path,
+                                         file_filter = path_filter,
+		                         shard_id = device_id,
+                                         num_shards = num_gpus)
+
         
+    def define_graph(self):
+        inputs = self.input(name="Reader")
+
+        return inputs
+
+    
+class NumpyReaderDynamicPipeline(Pipeline):
+    def __init__(self, path, batch_size, path_filter="*.npy",
+                 num_threads=1, device_id=0, num_gpus=1):
+        super(NumpyReaderDynamicPipeline, self).__init__(batch_size,
+                                                         num_threads,
+                                                         device_id)
+
+        self.rng = np.random.RandomState(12345)
+        self.get_anchor = ops.ExternalSource()
+        self.get_shape = ops.ExternalSource()
         self.input = ops.NumpyReader(file_root = path,
                                      file_filter = path_filter,
                                      shard_id = device_id,
                                      num_shards = num_gpus)
-
         
     def define_graph(self):
-        inputs = self.input(name="Reader")
+        self.anchor = self.get_anchor()
+        self.shape = self.get_shape()
+        inputs = self.input(name="Reader",
+                            anchor=self.anchor,
+                            shape=self.shape)
 
-        return inputs
+        return inputs, self.anchor, self.shape
 
+    def iter_setup(self):
```

Comment:
Please use the new `ExternalSource` and forget about `iter_setup`.

Reply:
done!